src/main/java/com/treasure/hunt/strategy/hider/impl/GameField.java

```diff
@@ -0,0 +1,521 @@
+        Pair<Coordinate, Double> sampledOnEdge;
+        Pair<Coordinate, Double> calcOnEdge;
+
+        List<Pair<Coordinate, Double>> worstEdgePoints = new ArrayList<>();
+
+        for (int currentCoordinateIndex = 1; currentCoordinateIndex < edges.length; currentCoordinateIndex++) {
+            calcOnEdge = calcMaxConstantPointOnSegment(edges[currentCoordinateIndex - 1], edges[currentCoordinateIndex], this.currentPlayersPosition.getCoordinate());
+            worstEdgePoints.add(calcOnEdge);
+            sampledOnEdge = sampleMaxConstantPointOnLineSegment(edges[currentCoordinateIndex - 1], edges[currentCoordinateIndex], this.currentPlayersPosition.getCoordinate());
+            log.info("line (" + edges[currentCoordinateIndex - 1] + ", " + edges[currentCoordinateIndex] + ")");
+            log.info("Calc " + calcOnEdge);
+            log.info("Samp " + sampledOnEdge);
+            if (sampledOnEdge.getValue() > bestSampled.getValue()) {
+                bestSampled = sampledOnEdge;
+            }
+            if (calcOnEdge.getValue() > bestCalc.getValue()) {
+                bestCalc = calcOnEdge;
+                log.debug("new best Const " + bestCalc.getValue() + " at " + bestCalc.getKey() + " at line (" + edges[currentCoordinateIndex - 1] + ", " + edges[currentCoordinateIndex] + ")");
+            }
+        }
+
+        worstEdgePoints.sort(new Comparator<Pair<Coordinate, Double>>() {
+            @Override
+            public int compare(Pair<Coordinate, Double> coordinateDoublePair, Pair<Coordinate, Double> t1) {
+                return coordinateDoublePair.getValue().compareTo(t1.getValue());
+            }
+        }.reversed());
+        log.info("worst Point List:" + worstEdgePoints.size() + "first element: " + worstEdgePoints.get(0));
+        log.info("best overall Sampled constant is" + bestSampled.getValue() + " at " + bestSampled.getKey().toString());
+        log.info("best overall calculated constant is" + bestCalc.getValue() + " at " + bestCalc.getKey().toString());
+        return worstEdgePoints;
+    }
+
+    /**
+     * Alternate to {@sampleMaxConstantPointOnLineSegment}
+     * Calculates the Point on the boundarySegment described by p1,p2 of the possible Area which results in the largest Constant of PathMin * C = PathActual
+     *
+     * @param p1     the Start of the Segment v
+     * @param p2     the End of the Segment v
+     * @param player current Position of the Player
+     * @return the Point and its corresponding constant C
+     */
+    private Pair<Coordinate, Double> calcMaxConstantPointOnSegment(Coordinate p1, Coordinate p2, Coordinate player) {
+        LineSegment unscaled = new LineSegment(p1, p2);
+        //log.debug("Calculating on " + unscaled);
```

Comment:
... Well if u want you can leave this log statements just make them trace  if u want them to not normally show up